holoviews/core/traversal.py

```diff
@@ -89,6 +89,11 @@ def unique_dimkeys(obj, default_dim='Frame'):
             if not matches:
                 unique_keys.append(padded_key)
 
+    # Add cartesian product of DynamicMap values to keys
+    values = [d.values for d in all_dims]
+    if obj.traverse(lambda x: x, ['DynamicMap']) and values and all(values):
+        unique_keys += list(zip(*cartesian_product(values)))
+
     with item_check(False):
         sorted_keys = NdMapping({key: None for key in unique_keys},
                                 kdims=all_dims).data.keys()
```

Comment:
Why do we do this? Is this more information than taking the set of keys then sorting them?

Reply:
There used to be at least, not sure its the case anymore though. Might leave a not to simplify this.

Reply:
Actually I'll open an issue because this is related to sorting issues.

Reply:
Opened the issue here: https://github.com/ioam/holoviews/issues/1835